omc3/correction/handler.py

```diff
@@ -109,6 +116,49 @@ def correct(accel_inst: Accelerator, opt: DotDict) -> None:
     LOG.info("Finished Iterative Global Correction.")
 
 
+
+def _update_response(
+    accel_inst: Accelerator, 
+    corrected_model: pd.DataFrame,
+    corrected_elements: pd.DataFrame,
+    delta: pd.DataFrame, 
+    optics_params: Sequence[str], 
+    variable_categories: Sequence[str], 
+    update_response: bool | str,
+    ) -> dict[str, pd.DataFrame]:
+    """ Create an updated response matrix.
+    
+    If we are to compute the response including the DPP, then we have to do so from MAD-X, 
+    as we do not have the analytical formulae.
+    Otherwise we go through the way of computing the response the user requested.
+
+    In case of dpp being modified in the correction, we have to add the new dpp value
+    to the accelerator instance, to run the `twiss` in madx at this dpp.
+    All other parameters are taken care of in the model/elements.
+    """
+    # update model by creating a copy of the accelerator instance
+    accel_inst_cp = copy.deepcopy(accel_inst)
+    accel_inst_cp.model = corrected_model
+    accel_inst_cp.elements = corrected_elements
```

Comment:
My issue with why I didn't do that, even though it is much nicer, is that the `elements` can be quite large and I was afraid that `deepcopy` will then double the memory footprint. We can leave it like that but we should be aware of this in case we ever see issues. On the other hand, we basically never update the response anyway and maybe this is all negligable ^^

Reply:
Perhaps `copy.replace(obj, /, **changes)` would be better here? This would prevent elements being copied, as we're replacing it?